orttraining/orttraining/training_api/include/module.h

```diff
@@ -91,13 +91,23 @@ struct Module {
   // Returns the output count for eval graph
   size_t GetEvalModeOutputCount() const noexcept;
 
+  // Return size of all parameters
+  size_t GetParametersSize(const bool trainable_only=true) const noexcept;
+
+  // Copy parameters onto contiguous buffer held by parameters_buffer
+  Status CopyParametersToBuffer(OrtValue& parameters_buffer, const bool trainable_only=true);
+  
+  // Copy parameter values from contiguous buffer held by parameters_buffer onto parameters
+  Status CopyBufferToParameters(OrtValue& parameters_buffer, const bool trainable_only=true);
+
  private:
   std::unique_ptr<onnxruntime::InferenceSession> train_sess_{nullptr};
   std::unique_ptr<onnxruntime::InferenceSession> eval_sess_{nullptr};
   std::vector<std::string> train_input_names_;
   std::vector<std::string> train_output_names_;
   std::vector<std::string> eval_input_names_;
   std::vector<std::string> eval_output_names_;
+  std::vector<std::string> weight_names_;
```

Comment:
We need to maintain the order because we need to get/set the parameter from/to the buffer in the same order. Changing the way parameters are held is not a dealbreaker, and can be done in a separate PR as required